### PR TITLE
better error message for assertAll, new test style, #1379

### LINKED ISSUE #1379: allSatisfy for maps should report all failing entries
#### Summary

A similar improvement was done for Iterable, it should be available for maps too.

### PR DESCRIPTION
**Pull request by almondtools**
#### Check List:
* Fixes #1379 
* Unit tests : YES (Updates)
* Javadoc with a code example (API only) : NA

### CODE REVIEW

File: src/test/java/org/assertj/core/internal/maps/Maps_assertAllSatisfyingConsumer_Test.java [hunk 1]
```diff
@@ -50,36 +54,47 @@ public void should_pass_if_all_entries_satisfy_the_given_requirements() {
...
-    failBecauseExpectedAssertionErrorWasNotThrown();
+    // WHEN
+    AssertionError error = expectAssertionError(() -> maps.assertAllSatisfy(someInfo(), greatPlayers, (team, player) -> {
+      assertThat(team).isIn("Lakers", "Bulls", "Spurs");
+      assertThat(player.getPointsPerGame()).as("%s %s ppg", player.getName().first, player.getName().getLast())
+                                           .isGreaterThanOrEqualTo(30);
+    }));
+
+    // THEN
+    assertThat(error).isNotNull();
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/internal/maps/Maps_assertAllSatisfyingConsumer_Test.java]: can be removed as `expectAssertionError` already checks that

File: src/test/java/org/assertj/core/internal/maps/Maps_assertAllSatisfyingConsumer_Test.java [hunk 2]
```diff
@@ -50,36 +54,47 @@ public void should_pass_if_all_entries_satisfy_the_given_requirements() {
...
-      assertThat(player.getPointsPerGame()).isGreaterThan(200);
+
+    // WHEN THEN
+    maps.assertAllSatisfy(someInfo(), greatPlayers, ($1, $2) -> {
+      assertThat(true).isFalse();
     });
   }
 
   @Test
   public void should_fail_if_one_entry_does_not_satisfy_the_given_requirements() {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/internal/maps/Maps_assertAllSatisfyingConsumer_Test.java]: now that we report all the errors I think we should have two `should_fail` tests, one making sure it fails when only one entry does not satisfy the requirements (named like this test) and another that verifies we report all the errors (basically the body of that test to be renamed as `should_report_all_the_entries_not_satisfying_the_given_requirements`

File: src/main/java/org/assertj/core/internal/Maps.java [hunk 3]
```diff
@@ -94,8 +99,23 @@ public static Maps instance() {
...
     checkNotNull(entryRequirements, "The BiConsumer<K, V> expressing the assertions requirements must not be null");
     assertNotNull(info, actual);
-    actual.entrySet().stream()
-          .forEach(entry -> entryRequirements.accept(entry.getKey(), entry.getValue()));
+
+    List<UnsatisfiedRequirement> unsatisfiedRequirements = actual.entrySet().stream()
+                                                                 .map(entry -> failsRequirements(entryRequirements, entry))
+                                                                 .filter(Optional::isPresent)
+                                                                 .map(Optional::get)
+                                                                 .collect(toList());
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/internal/Maps.java]: nice!

**Comment by joel-costigliola** [comment]
@almondtools  reviewed!

**Comment by almondtools** [comment]
you may continue.

**Comment by joel-costigliola** [comment]
Integrated thanks @almondtools!

**Pull request closed without merging**